jwst/background/background_sub.py

```diff
@@ -26,85 +26,171 @@ def background_sub(input_model, bkg_list, sigma, maxiters):
     bkg_list: filename list
         list of background exposure file names
 
+    sigma: float, optional
+        Number of standard deviations to use for both the lower
+        and upper clipping limits.
+
+    maxiters: int or None, optional
+        Maximum number of sigma-clipping iterations to perform
+
+    bkg_model: JWST data model
+        background data model
+
     Returns
     -------
     result: JWST data model
         background-subtracted target data model
 
     """
+    # Subtract the average background from the member
+    log.debug(' subtracting avg bkg from {}'.format(input_model.meta.filename))
 
     # Compute the average of the background images associated with
     # the target exposure
-    bkg_model = average_background(bkg_list, sigma, maxiters)
+    if input_model.meta.model_type == "ImageModel":
+        data_dim = 2  # rate
+    elif input_model.meta.model_type == "CubeModel":
+        data_dim = 3  # rateints
+    else:
+        log.error("Input target exposure must be either a CubeModel or an ImageModel.")
+        raise ValueError("Expected either a CubeModel or an ImageModel.")
 
-    # Subtract the average background from the member
-    log.debug(' subtracting avg bkg from {}'.format(input_model.meta.filename))
-    result = subtract_images.subtract(input_model, bkg_model)
+    bkg_model = average_background(bkg_list, sigma, maxiters, data_dim)
+
+    if data_dim == 2:  # rate
+        result = subtract_images.subtract(input_model, bkg_model)
+    elif data_dim == 3:  # rateints
```

Comment:
Don't need a separate (new) function to handle 3D data. Just call the original `subtract` function as written. The numpy array arithmetic at https://github.com/spacetelescope/jwst/blob/master/jwst/background/subtract_images.py#L32 will automatically subtract the 2D avg bkg from each integration of the input data if the input is 3D.

Reply:
Resolved this.